cmd/influx/cli/cli.go

```diff
@@ -73,6 +77,22 @@ func New(version string) *CommandLine {
 
 // Run executes the CLI
 func (c *CommandLine) Run() error {
+	// If we are not running in an interactive terminal, read stdin completely
+	// and execute a query. Do not allow meta commands.
+	if !c.ForceTTY && !terminal.IsTerminal(int(os.Stdin.Fd())) {
+		if err := c.Connect(""); err != nil {
+			return fmt.Errorf(
+				"Failed to connect to %s\nPlease check your connection settings and ensure 'influxd' is running.",
+				c.Client.Addr())
+		}
+
+		cmd, err := ioutil.ReadAll(os.Stdin)
```

Comment:
We need to send the entire thing as part of the query with the current client library so I don't know if we really have a choice. Using a scanner and reading line by line won't really work since we want to send all of it as the same query and we can also split commands between multiple lines. Semi-colons are what we use as delimiters.

I do think we can possibly improve it in the future. I added support for sending the query as a file, but we don't currently use that in any of the clients.